handshake_server.go

```diff
@@ -69,6 +69,7 @@ func (c *Conn) serverHandshake() error {
 	// For an overview of TLS handshaking, see https://tools.ietf.org/html/rfc5246#section-7.3
 	// and https://tools.ietf.org/html/draft-ietf-tls-tls13-18#section-2
 	c.buffering = true
+	c.clientHello = hs.clientHello.marshal()
```

Comment:
What about moving this after line 67, just after successfully reading a CH?

Reply:
yup, why not. I'll do that

Reply:
I would personally have kept this reference. The handshake is not trivial and having a quick overview of the steps would help others who are not TLS experts to understand this code and verify its correctness.